pkg/storagepolicy/sdkstoragepolicy.go

```diff
@@ -87,82 +98,92 @@ func (p *SdkPolicyManager) Create(
 	ctx context.Context,
 	req *api.SdkOpenStoragePolicyCreateRequest,
 ) (*api.SdkOpenStoragePolicyCreateResponse, error) {
-	if req.StoragePolicy.GetName() == "" {
+	if req.GetStoragePolicy().GetName() == "" {
 		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
-	} else if req.StoragePolicy.GetPolicy() == nil {
+	} else if req.GetStoragePolicy().GetPolicy() == nil {
 		return nil, status.Error(codes.InvalidArgument, "Must supply Volume Specs")
 	}
 
+	// Add ownership details to storage policy
+	// user who has RBAC access to storage policy can create one
+	req.StoragePolicy.Ownership = api.OwnershipSetUsernameFromContext(ctx, req.GetStoragePolicy().GetOwnership())
+
 	// Since VolumeSpecPolicy has oneof method of proto,
-	// we need to marshal it into string using protobuf jsonpb
+	// we need to marshal sdkStoragePolicy into string using protobuf jsonpb
 	m := jsonpb.Marshaler{OrigName: true}
-	policyStr, err := m.MarshalToString(req.StoragePolicy.GetPolicy())
+	policyStr, err := m.MarshalToString(req.GetStoragePolicy())
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Json Marshal failed for policy %s: %v", req.StoragePolicy.GetName(), err)
+		return nil, status.Errorf(codes.Internal, "Json Marshal failed for policy %s: %v", req.GetStoragePolicy().GetName(), err)
 	}
 
-	_, err = p.kv.Create(prefixWithName(req.StoragePolicy.GetName()), policyStr, 0)
+	_, err = p.kv.Create(prefixWithName(req.GetStoragePolicy().GetName()), policyStr, 0)
 	if err == kvdb.ErrExist {
-		return nil, status.Errorf(codes.AlreadyExists, "Storage Policy already exist : %v", req.StoragePolicy.GetName())
+		return nil, status.Errorf(codes.AlreadyExists, "Storage Policy already exist : %v", req.GetStoragePolicy().GetName())
 	} else if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to save storage policy: %v", err)
 	}
 
+	logrus.Infof("Storage Policy %v is created", req.GetStoragePolicy().GetName())
 	return &api.SdkOpenStoragePolicyCreateResponse{}, nil
 }
 
 // Update Storage policy
+// TODO: Decide whether update storage policy request should update ownership at all
+// We can check if user is admin then only allow update ownership
 func (p *SdkPolicyManager) Update(
 	ctx context.Context,
 	req *api.SdkOpenStoragePolicyUpdateRequest,
 ) (*api.SdkOpenStoragePolicyUpdateResponse, error) {
-	if req.StoragePolicy.GetName() == "" {
+	if req.GetStoragePolicy().GetName() == "" {
 		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
 	}
 
-	if req.StoragePolicy.GetPolicy() == nil {
+	if req.GetStoragePolicy().GetPolicy() == nil {
 		return nil, status.Error(codes.InvalidArgument, "Must supply Volume Specs")
 	}
 
-	// Get Existing details to merge
-	pol, err := p.Inspect(ctx,
+	// Get Existing details to check access
+	oldPolicy, err := p.Inspect(ctx,
```

Comment:
This checks read access, but it should then check for WRITE access in Update.

Reply:
oh i see. instead of doing it at line 162, it should be done right after this inspect.